### PR TITLE
Fundamentals block - main and optional tasks

### PR DESCRIPTION
**Pull request by lerik14**

### CODE REVIEW

File: java_fundamentals_block/main_task/src/main/java/Actions.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                int number = Integer.parseInt(input);
+                sum += number;
+                multiple *= number;
+            }
+            else {
+                break;
+            }
+        }
+        System.out.println("The sum of all numbers is " + sum);
+        System.out.println("The product of all numbers is " + multiple);
```
Comment (Orkyboy) [inline java_fundamentals_block/main_task/src/main/java/Actions.java:62]: product? is it typo?

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        for (int i = 1; i < array.length; i++) {
+            if (array[i] > longElem) {
+                longElem = array[i];
+            }
+        }
+        System.out.println("The longest element is " + longElem);
+        System.out.println("The length of this element is " + elemLength(longElem));
+    }
+
+    public static void sort(int[] array) {
```
Comment (Orkyboy) [inline java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: rename to sortByLength to emphasize what function does

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class OptionalTask1 {
+
+    private static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static int[] inputElem() throws IOException {
```
Comment (Orkyboy) [inline java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: i suggest renaming to getElementsArrayFromConsole

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        int numberOfElements = Integer.parseInt(reader.readLine());
+        int[] array = new int[numberOfElements];
+        System.out.println("Enter the elements:");
+        for (int i = 0; i < numberOfElements; i++) {
+            array[i] = Integer.parseInt(reader.readLine());
+        }
+        return array;
+    }
+
+    public static void findShortElement(int[] array) {
```
Comment (Orkyboy) [inline java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: I think in current implementation these functions do 2 things:
1. action (like find shortest element here)
2. printing to console

I suggest leave only single responsibility to the function and move printing out to `OptionalTaskMain.java`

so converted this function will be like
```    
public static int findShortestElement(int[] array) {
        int shortElem = array[0];
        for (int i = 1; i < array.length; i++) {
            if (array[i] < shortElem) {
                shortElem = array[i];
            }
        }
   return shortElem;
}
```

And then in OptionalTaskMain
```
int shortestElem = OptionalTask1.findShortElement(array);
        System.out.println("The longest element is: " + shortestElem);
        System.out.println("The length of this element is: " + OptionalTask1.elemLength(shortestElem));
```

File: java_fundamentals_block/main_task/src/main/java/Actions.java (hunk 1 above)
Comment (lerik14) [reply java_fundamentals_block/main_task/src/main/java/Actions.java:62]: No. The product of numbers is the result you get when you multiply them together.

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java (hunk 2 above)
Comment (lerik14) [reply java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: done

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java (hunk 3 above)
Comment (lerik14) [reply java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: good idea. Done

File: java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java (hunk 4 above)
Comment (lerik14) [reply java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java]: Done for this and other methods

**Review by Orkyboy — APPROVED** [review]

**Pull request merged**